### DISCUSSION TITLE
Weather Balloon script crash

### DISCUSSION THREAD

**Discussion by dpculp** [JSBSim-Team/jsbsim#643] [General]
When executing the script weather-balloon.xml as supplied by the repo I get a core dump "Floating point exception" after about 4980 seconds of virtual time.  This is at the point in the flight around 92K feet where the balloon is deflating and beginning descent.

I changed the termination event to stop execution at 80K feet of altitude and the script ran perfectly.

It seems the transition from ascent to descent is not working presently in the weather balloon (gas cell) model.  I'll investigate further.

**Comment by dpculp** [comment]
Some further details on the balloon burst, and code formatting practice for me :)  I added a notification of the burst event.

```
Repeating Notify (Event 1) executed at time: 4920.006365
    position/h-agl-ft = 91043.379714
    velocities/vt-fps = 25.095454
    metrics/radius-ft = 10.821114


Burst (Event 2) executed at time: 4979.506341
    position/h-agl-ft = 92551.476349


Repeating Notify (Event 1) executed at time: 4980.006341
    position/h-agl-ft = 92563.883070
    velocities/vt-fps = 23.461809
    metrics/radius-ft = 9.973560

Floating point exception (core dumped)
```

**Reply by bcoconni** [reply]
Confirmed ! The script `scripts/weather-balloon.xml` crashes at 92 kft although the simulated time at which this occurs on my test is lower than yours (4260s vs 4980s):
```text
Repeating Notify (Event 1) executed at time: 4200.006653
    position/h-agl-ft = 90584.452911
    velocities/vt-fps = 25.274223
    metrics/radius-ft = 10.742502


Repeating Notify (Event 1) executed at time: 4260.006629
    position/h-agl-ft = 92098.692025
    velocities/vt-fps = 25.298340
    metrics/radius-ft = 11.003866

Exception en point flottant (core dumped)
```

**Reply by bcoconni** [reply]
I have created the issue #654 to log the issue as a bug.
